### ISSUE TITLE
Implement real-time drum separation lambda

### ISSUE THREAD

**Issue by harryaskham** [harryaskham/pulseaudio-lambda#26]
### Objective
Implement the actual drum separation lambda using the selected ML model and framework.

#### Model Integration:
- [ ] Load pre-trained drum separation model
- [ ] Optimize model for inference speed
- [ ] Implement model-specific preprocessing/postprocessing
- [ ] Handle model warmup and initialization

#### Lambda Implementation:
- [ ] Create `lambdas/drumless-ml.py` 
- [ ] Integrate with ML framework from issue #25
- [ ] Implement configurable separation modes:
  - `drums_only` - Extract just drums
  - `no_drums` - Remove drums (default)
  - `both` - Output both streams somehow
- [ ] Add intensity/threshold controls

#### Real-time Optimization:
- [ ] Optimize chunk size for latency vs quality
- [ ] Implement overlapping window processing
- [ ] GPU memory pool management
- [ ] Asynchronous processing pipeline

#### Performance Tuning:
- [ ] Profile bottlenecks in processing pipeline
- [ ] Optimize audio format conversions
- [ ] Minimize Python overhead
- [ ] Implement caching for repeated patterns

#### Lambda Interface:
```bash
# Usage examples:
./pulseaudio-lambda "lambdas/drumless-ml.py"
./pulseaudio-lambda "lambdas/drumless-ml.py --mode=no_drums --intensity=0.8"
./pulseaudio-lambda "lambdas/drumless-ml.py --model=demucs --device=cuda"
```

#### Configuration Options:
- [ ] Model selection (if multiple supported)
- [ ] Processing intensity/threshold
- [ ] GPU device selection
- [ ] Quality vs speed trade-offs

### Acceptance Criteria:
- Working real-time drum separation
- Latency < 100ms end-to-end (stretch: <50ms)
- Quality better than sox-based approach
- Compatible with existing pulseaudio-lambda interface
- Configurable separation parameters

### Performance Requirements:
- Process 44.1kHz stereo audio in real-time
- <2GB GPU memory usage
- Graceful degradation if GPU unavailable
- No audio dropouts or glitches

### Dependencies:
- Completed issue #24 (model selection)
- Completed issue #25 (ML framework)
- GPU-enabled development environment

### Testing:
- [ ] A/B comparison with sox drumless filters
- [ ] Latency measurement and optimization
- [ ] Long-running stability testing
- [ ] Various music genre testing
- [ ] CPU fallback testing
